### PR TITLE
feat: add docker instructions in readme

### PR DESCRIPTION
**Pull request by kshitij-k-osmosys**
## Description

- Update readme to include steps for setting the app in docker.
- Updated docker-compose variables. 

## Summary by CodeRabbit

- **Documentation**
	- Updated setup instructions for running the application in a Docker environment on Linux, including steps for Docker installation, project cloning, setting environment variables, building containers, and accessing the application.
	- Added details on accessing Swagger UI and testing APIs in the setup guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,41 @@
 - Converts an HTML document to PDF.
 - Replace placeholders in the document with actual string data.
 
-# How to setup (Windows)
+# How to setup app in a Docker based environment (Linux)
+
+The development of this application is done on Visual Studio IDE on a Windows machine. Setting up the app in a Docker based environment enables developers of non-Windows origins to run the backend application on their machine to test the APIs.
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 2]
```diff
@@ -16,7 +16,41 @@
...
+
+## Steps
+
+1. [Install Docker](https://docs.docker.com/engine/install/) on your machine. Choose to follow the instructions based on your device OS.
+2. [Install Docker Compose](https://docs.docker.com/compose/install/). A separate installation is required for Linux based OS. If you are using Windows or MacOS, installing the Docker Desktop app includes Docker Compose.
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 3]
```diff
@@ -16,7 +16,41 @@
...
+      - BUILD_CONFIGURATION=Release
+```
+
+6. Ensure Docker is running. Then, in root directory of project, execute the following command to build container for `document-service`:
+
+```shell
+docker-compose -f docker-compose.yaml up
+```
+
+7. Project will run on `http://localhost:5000`.
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 4]
```diff
@@ -16,7 +16,41 @@
...
+8. You can access the **Swagger UI** at `http:localhost:5000/swagger/index.html` in **Development Enviroment**.
+9. Test the API via **Postman**. The app can be accessed using `http://localhost:5000/<API>`.
+
+# How to setup library (Windows)
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -16,7 +16,41 @@ DocumentService is a library with the following functions
...
 
 ## Steps for installing wkhtmltopdf
 - Go to website: https://wkhtmltopdf.org/downloads.html
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 6]
```diff
@@ -16,7 +16,41 @@
...
+```
+
+6. Ensure Docker is running. Then, in root directory of project, execute the following command to build container for `document-service`:
+
+```shell
+docker-compose -f docker-compose.yaml up
+```
+
+7. Project will run on `http://localhost:5000`.
+8. You can access the **Swagger UI** at `http:localhost:5000/swagger/index.html` in **Development Enviroment**.
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 7]
```diff
@@ -16,7 +16,41 @@
 - Converts an HTML document to PDF.
 - Replace placeholders in the document with actual string data.
 
-# How to setup (Windows)
+# How to setup app in a Docker based environment (Linux)
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 8]
```diff
@@ -10,13 +10,47 @@ DocumentService is a library with the following functions
...
+- Replace images with image's placeholder. The image's position will be maintained based on the position of its placeholder image. Image size will also be maintained based on placeholder image.
 
 ## PDF document generation
 - Converts an HTML document to PDF.
 - Replace placeholders in the document with actual string data.
 
-# How to setup (Windows)
+# How to set up the Application in a Docker-based environment (Linux)
+
+The development of this application is done on Visual Studio IDE on a Windows machine. Setting up the app in a Docker-based environment enables developers of non-Windows origins to run the backend application on their machine to test the APIs.
```
Comment (sameer-s-b) [inline README.md:21]: > "The development of this application is done on Visual Studio IDE on a Windows machine"
This information is irrelevant.

Comment (kshitij-k-osmosys) [reply README.md]: Removed this line

**Review by sameer-s-b — APPROVED** [review]

**Pull request merged**